### PR TITLE
planner: set the partition id to table id in DAG request

### PR DESCRIPTION
**Pull request by lzmhhh123**
### What problem does this PR solve? 
When the `tablescan` and `indexscan` convert to dag request, the field `TableID` is always set as `tableinfo.ID`, it doesn't matter for TiKV. Because TiKV scans key ranges. But it's not always correct for TiFlash(schemaful). When approaching to the partition table, it should be `partitionID`.

### What is changed and how it works?
Set the `TableID` as `physicalTableID` when it's a partition table.

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes

 - Has exported function/method change

### CODE REVIEW

**Comment by Rustin-Liu** [comment]
@sre-bot /run-all-tests

File: planner/core/plan_to_pb.go [hunk 1]
```diff
@@ -95,6 +95,9 @@ func (p *PhysicalTableScan) ToPB(ctx sessionctx.Context) (*tipb.Executor, error)
 		Columns: model.ColumnsToProto(p.Columns, p.Table.PKIsHandle),
 		Desc:    p.Desc,
 	}
+	if p.isPartition {
+		tsExec.TableId = p.physicalTableID
```
Comment (crazycs520) [inline planner/core/plan_to_pb.go]: I have a question, `TiFlash` doesn't use the `kv range`? since the `kv range` encoded with physical table ID.

Comment (lzmhhh123) [reply planner/core/plan_to_pb.go]: It's schemaful. TiFlash decode column with the table schema. So it needs a correct table id.

**Pull request merged**